src/examples/IK/exampleIK.cpp

```diff
@@ -273,177 +262,137 @@ int main()
+
+    matioCpp::Struct human_state = human_data("human_state").asStruct(); // Access structure "human_state"  and convert it to a structure
+    matioCpp::Struct base_position = human_state("base_position").asStruct(); // Access struct "base_position" and convert it to a struct
     matioCpp::MultiDimensionalArray<double> basePos_data
-        = base_position("data").asMultiDimensionalArray<double>(); // Access multidimensional array
-                                                                   // "data" within "base_position"
-                                                                   // structure and convert it to a
-                                                                   // multidimensional array of type
-                                                                   // double
-    matioCpp::Struct base_orientation
-        = human_state("base_orientation").asStruct(); // Access structure "base_orientation" within
-                                                      // "human_state" structure and convert it to a
-                                                      // structure
+        = base_position("data").asMultiDimensionalArray<double>(); // Access multidimensional array "data" within "base_position"
+                                                                   // structure and convert it to a multidimensional array of type double
+
+    // Retrieve baseOrientation_data -->[quaternion, all timestamps]
+
+    matioCpp::Struct base_orientation = human_state("base_orientation").asStruct(); // Access structure "base_orientation" within
+                                                                                    // "human_state" structure and convert it to a structure
     matioCpp::MultiDimensionalArray<double> baseOrientation_data
-        = base_orientation("data").asMultiDimensionalArray<double>(); // Access multidimensional
-                                                                      // array "data" within
-                                                                      // "base_orientation"
-                                                                      // structure and convert it to
-                                                                      // a multidimensional array of
-                                                                      // type double
+        = base_orientation("data").asMultiDimensionalArray<double>(); // Access multidimensional array "data" within "base_orientation"
+                                                                      // structure and convert it to a multidimensional array of type double
+
+    //*************************************************************************************************************************************
+
+    // Get the dimension of the data (total number of timestamps) from Matlab file
+    matioCpp::File file("/path/to/matlab1.mat"); // Create a file object to read data from MATLAB file
```

Comment:
And here the same.

Reply:
Done with commit https://github.com/ami-iit/biomechanical-analysis-framework/pull/31/commits/575da1ffecd6145855cad2d5019c3e90325b45b3